src/Spice86.Core/Emulator/CPU/CfgCpu/Feeder/CfgNodeFeeder.cs

```diff
@@ -13,21 +13,21 @@ namespace Spice86.Core.Emulator.CPU.CfgCpu.Feeder;
 /// Handles coherency between the memory and the graph of instructions executed by the CPU.
 /// Next node to execute is normally the next node from the graph but several checks are done to make sure it is really it:
 ///  - The node is not null (otherwise it is taken from memory)
-///  - If the node is an assembly node, it is the same as what is currently in memory, otherwise it means self modifying code is beeing detected
-///  - If self modifying code is beeing detected, Discriminator node is beeing injected instead.
+///  - If the node is an assembly node, it is the same as what is currently in memory, otherwise it means self modifying code is being detected
+///  - If self modifying code is being detected, Discriminator node is being injected instead.
 /// Once the node to execute is determined, it is linked to the previously executed node in the execution context if possible.
 /// </summary>
 public class CfgNodeFeeder {
     private readonly State _state;
     private readonly InstructionsFeeder _instructionsFeeder;
-    private readonly NodeLinker _nodeLinker = new();
+    private readonly NodeLinker _nodeLinker;
     private readonly DiscriminatorReducer _discriminatorReducer;
 
-    public CfgNodeFeeder(IMemory memory, State state, MachineBreakpoints machineBreakpoints) {
+    public CfgNodeFeeder(InstructionsFeeder instructionsFeeder, DiscriminatorReducer discriminatorReducer, NodeLinker nodeLinker,  State state) {
         _state = state;
-        _instructionsFeeder = new(machineBreakpoints, memory, state);
-        _discriminatorReducer = new(new List<IInstructionReplacer<CfgInstruction>>()
```

Comment:
I am not a fan of moving creation of this object with logic quite specific to this domain outside of it (the 2 objects list it takes into parameter)

Reply:
Every class has domain specific classes. For example DOS has the DOSFileManager, the ConsoleDevice, the Printer, StdAux, the DOSPathResolver, DOSMemoryManager, etc... Yet they are all instantiated outside of DOS now. Which led to simplified code in the DOSINT2FHandler for example. Null Reference Types question marks and ifs where removed, because the exact dependency needed was directly provided to the class.

From the point of view of IoC, whether they are domain specific or not does not matter. What is important is that they are dependencies, and dependencies should be injected, not created.

The goal is to get:

- a centralized place for object creation and object lifetime management in Program.cs / ServiceCollectionExtension.cs. Being centralized, it will make dependency management easier
- Easier testing
- Simpler code (sometimes)

Reply:
You're right. Those classes should remain internal.

Reply:
This has been reverted.